pkg/proxy/iptables/proxier.go

```diff
@@ -864,9 +902,16 @@ func endpointsToEndpointsMap(endpoints *api.Endpoints, hostname string) proxyEnd
 					glog.Warningf("ignoring invalid endpoint port %s with empty host", port.Name)
 					continue
 				}
+				// Filter out the incorrect IP version case.
+				if !utilproxy.CheckIPVersion(addr.IP, ecm.isIPv6Mode) {
```

Comment:
Maybe easier to read:

```
if utilproxy.IsIPv6(addr.IP) != ecm.isIPv6Mode {
```

Same at other callsites

Reply:
Good point, suggestion applied.